pypeit/scripts/rectify_2dspec.py

```diff
@@ -0,0 +1,214 @@
+                                                                        wave_grid_min=wmin, wave_grid_max=wmax)
+
+                # Loop over slits and rectify each one
+                imgrect_list = []
+                for slitidx, slit_id in enumerate(slit_ids):
+                    this_mask = (slitmask == slit_id)
+                    # check if this slit was masked. If so, skip it.
+                    slitord_id = slitord_ids[slitidx]
+                    if not np.any(this_mask):
+                        msgs.warn(f'Slit/order {slitord_id} on {detname} is fully masked. Skipping it.')
+                        continue
+                    msgs.info(f'Rectifying slit/order {slitord_id}')
+
+                    slit_cen = spec2d.slits.center[:,slitidx]
+                    mask = spec2d.bpmmask.mask == 0
+
+                    imgrect_dict = coadd.compute_coadd2d([slit_cen], [spec2d.sciimg],
+                                                         [spec2d.ivarmodel],[spec2d.skymodel],
+                                                         [mask],[this_mask],
+                                                         [spec2d.waveimg], wave_grid)
+                    imgrect_list.append(imgrect_dict)
+
+
+                # Get dimensions for each slit
+                nspat_vec = np.array([cdict['nspat'] for cdict in imgrect_list])
+                nspec_rect = len(wave_grid_mid)
+                nspat_rect = int(np.sum(nspat_vec) + (spec2d.slits.nslits + 1) * pad)
+
+                # Initialize output array
+                # this is the rectified image on a common wavelength grid for all slits
+                image_rect = np.zeros((nspec_rect, nspat_rect))
+                # the outputs below are not really used, but are useful for double-checking
+                # this is the wave image created by placing the rectified wavelengths for each slit in
+                # their proper location in the rectified image
+                waveimg_rect = np.zeros((nspec_rect, nspat_rect))
+                # this is the common wave image for the whole rectified image, created by repeating
+                # the wavelength grid for each spatial pixel
+                wavegrid_image = np.repeat(wave_grid_mid[:, np.newaxis], nspat_rect, axis=1)
+
+                # Loop over the rectified slits and place them in the output arrays
+                spat_left = pad
+                for islit, imgrect_dict in enumerate(imgrect_list):
+                    spat_righ = spat_left + nspat_vec[islit]
+
+                    # Get wavelength array for this slit
+                    wave_slit = imgrect_dict['wave_mid']
+                    nspec_slit = len(wave_slit)
+                    # get the spectral pixel where the slit should start/stop
+                    _wave_grid_mid = np.round(wave_grid_mid, 4)
+                    _wave_slit = np.round(imgrect_dict['wave_mid'], 4)
+                    wstart = np.where(_wave_grid_mid == _wave_slit[0])[0][0]
```

Comment:
I went back and forward between using `==` or `np.close` here, and at the end I decided to run 
`pypeit_rectify_2dspec PypeIt-development-suite/REDUX_OUT/*/*/*/spec2d*.fits`
That failed pretty soon for `np.close`, instead it was successful for all Dev Suite spec2d with `==`